src/test/java/htsjdk/tribble/index/IndexFactoryTest.java

```diff
@@ -112,4 +117,45 @@ public void testCreateTabixIndexOnBlockCompressed() {
                     "Tabix indexed (bgzipped) VCF does not contain sequence: " + samSequenceRecord.getSequenceName());
         }
     }
+
+    @Test
+    public void testWriteIndex() throws Exception {
```

Comment:
I changed exactly the same method (with a data provider instead) to `IndexTest`, because I think that it could live in that package (it is testing the interface method for different implementations).

I added a check for the sequence names and properties loaded and written, and a test bed file too.